int32 id
int32 type
point coord
bool is_running
bool done